### ISSUE TITLE
🐛 [BUG] - Image as picture does not use the "height" prop

### ISSUE THREAD

**Issue by seangolob** [primer/brand#604]
### Describe the bug

https://primer.style/brand/components/Image 

I am trying to create a consistent height of images while leveraging `sources`.

However, when I pass `as="picture"`, the height is not applied to the image. 

I beliave it is due to this line of code: https://github.com/primer/brand/blob/d38e44101b3a2b8031c0c68031b2ec0c66cff252/packages/react/src/Image/Image.tsx#L84

Is `height={width ?? '100%'}` intentional? 

![Image](https://github.com/primer/brand/assets/8423937/4d93b64e-32b6-4af4-b0b6-41a3ae1a19ce)

### Reproduction steps

Implement an Image like so:

```
<Card.Image
 as="picture"
 src={`https://placholder.com`}
 alt={'alt image'}
 sources={getSources('placeholder.com')} 
 height={328}
/>
```

### Expected behavior

I would expect the image to have a height of 328px.

**Comment by joshfarrant** [comment]
Hey @seangolob — Just a heads up that I'm looking into this for you.

I'm hoping to have a solution for you shortly. I've refactored the Image component in a WIP PR which should simplify things a bit and hopefully fix this issue.

**Comment by joshfarrant** [comment]
Hey again @seangolob 

That refactor PR is still under review, but I've put a quick fox together for the issue you were running into

https://github.com/primer/brand/pull/630

This will go out in the next release of Primer Brand which we're expecting to be out before the end of the week

**Comment by rezrah** [comment]
This has shipped in the [`0.34.2` release.](https://github.com/primer/brand/releases/tag/%40primer%2Freact-brand%400.34.2)
